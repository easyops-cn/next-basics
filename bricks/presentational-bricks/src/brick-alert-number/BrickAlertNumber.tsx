import React from "react";
import {
  RuleProps,
  BrickConditionalDisplay,
} from "../brick-conditional-display/BrickConditionalDisplay";

interface BrickAlertNumberProps {
  alertNumber: number;
}

const rules: RuleProps[] = [
  {
    condition: { $eq: 0 },
    style: {
      backgroundColor: "var(--color-success)",
      color: "white",
    },
  },
  {
    condition: { $gte: 1 },
    style: {
      backgroundColor: "var(--color-warning)",
      color: "white",
    },
  },
];

export function BrickAlertNumber(
  props: BrickAlertNumberProps
): React.ReactElement {
  return (
    <BrickConditionalDisplay
      data={props.alertNumber}
      value={props.alertNumber}
      rules={rules}
      type={null}
    />
  );
}
