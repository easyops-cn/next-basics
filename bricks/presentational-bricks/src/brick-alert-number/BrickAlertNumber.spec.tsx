import React from "react";
import { shallow } from "enzyme";
import { BrickAlertNumber } from "./BrickAlertNumber";

describe("BrickAlertNumber", () => {
  it("should work, 0", () => {
    const wrapper = shallow(<BrickAlertNumber alertNumber={0} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should work, 1", () => {
    const wrapper = shallow(<BrickAlertNumber alertNumber={1} />);
    expect(wrapper).toMatchSnapshot();
  });
});
