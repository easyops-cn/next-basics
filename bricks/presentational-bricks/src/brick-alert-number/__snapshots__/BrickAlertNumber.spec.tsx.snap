// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrickAlertNumber should work, 0 1`] = `
<BrickConditionalDisplay
  data={0}
  rules={
    Array [
      Object {
        "condition": Object {
          "$eq": 0,
        },
        "style": Object {
          "backgroundColor": "var(--color-success)",
          "color": "white",
        },
      },
      Object {
        "condition": Object {
          "$gte": 1,
        },
        "style": Object {
          "backgroundColor": "var(--color-warning)",
          "color": "white",
        },
      },
    ]
  }
  type={null}
  value={0}
/>
`;

exports[`BrickAlertNumber should work, 1 1`] = `
<BrickConditionalDisplay
  data={1}
  rules={
    Array [
      Object {
        "condition": Object {
          "$eq": 0,
        },
        "style": Object {
          "backgroundColor": "var(--color-success)",
          "color": "white",
        },
      },
      Object {
        "condition": Object {
          "$gte": 1,
        },
        "style": Object {
          "backgroundColor": "var(--color-warning)",
          "color": "white",
        },
      },
    ]
  }
  type={null}
  value={1}
/>
`;
